clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -972,16 +975,28 @@ private Record nextFetchedRecord() {
         private List<ConsumerRecord<K, V>> fetchRecords(int maxRecords) {
             if (isFetched)
                 return Collections.emptyList();
+            if (nextInlineException != null) {
```

Comment:
> We preserve that behavior for now. That means we must modify this patch to not skip over failed messages.

@hachikuji I think we all know that the bug fixed in this patch is orthogonal to whether we skip over messages that cause failure. Previously you mentioned that you can submit a separate patch to fix this behavior which were introduced by yourself. Is there any reason that you changed your mind and think "we must modify this patch to not skip over failed message"?

This is quite frustrating for me as a contributor to have to wait for 23 days just to fix this simple bug. I hope we can resolve this faster and do it in a way that can encourage open source contribution.